request.js

```diff
@@ -921,7 +921,7 @@ Request.prototype.onRequestResponse = function (response) {
     })
 
     var responseContent
-    if (self.gzip) {
+    if (self.gzip && self.method !== 'HEAD') {
```

Comment:
Figuring out if the response is empty beforehand would be great, will think about it. I would rather not use `try/catch` if there is some other way to detect the condition.